spi/w5500_spi/w5500_spi.c

```diff
@@ -0,0 +1,161 @@
+   Ignores the magnetometer, that is left as a exercise for the reader.
+
+   This is taking to simple approach of simply reading registers. It's perfectly
+   possible to link up an interrupt line and set things up to read from the
+   inbuilt FIFO to make it more useful.
+
+   NOTE: Ensure the device is capable of being driven at 3.3v NOT 5v. The Pico
+   GPIO (and therefor SPI) cannot be used at 5v.
+
+   You will need to use a level shifter on the I2C lines if you want to run the
+   board at 5v.
+
+   Connections on Raspberry Pi Pico board and a generic wiznet board, other
+   boards may vary.
+   GPIO 2 (pin 4) SCK/spi0_sclk -> CLK on wiznet board
+   GPIO 3 (pin 5) MOSI/spi0_tx -> MOSIon wiznet board
+   GPIO 4 (pin 6) MISO/spi0_rx-> MISO on wiznet board
+   GPIO 5 (pin 7) Chip select -> NCS on wiznet board
+   GPIO 6 (pin 9) Reset/GPIO -> Reset on wiznet board
+   3.3v (pin 36) -> VCC on wiznet board
+   GND (pin 38)  -> GND on wiznet board
+
+   Note: SPI devices can have a number of different naming schemes for pins. See
+   the Wikipedia page at https://en.wikipedia.org/wiki/Serial_Peripheral_Interface
+   for variations.
+   The particular device used here uses the same pins for I2C and SPI, hence the
+   using of I2C names
+*/
+
+#define PIN_SCK  2
+#define PIN_MOSI 3
+#define PIN_MISO 4
+#define PIN_CS   5
+#define PIN_RST  6
+
+#define SPI_PORT spi0
+#define READ_BIT 0x00
+#define TCP_SOCKET 0
+
+wiz_NetInfo gWIZNETINFO = { .mac = {0x00,0x08,0xdc,0x78,0x91,0x71},
+                            .ip = {192,168,0,15},
+                            .sn = {255,255,255,255},
+                            .gw = {192,168,0,1},
+                            .dns = {168,126,63,1},
+                            .dhcp = NETINFO_STATIC};
+#define ETH_MAX_BUF 2048
+
+unsigned char ethBuf0[ETH_MAX_BUF];
+void print_network_information(void);
+
+
+static inline void cs_select() {
+    asm volatile("nop \n nop \n nop");
+    gpio_put(PIN_CS, 0);  // Active low
+    asm volatile("nop \n nop \n nop");
+}
+
+static inline void cs_deselect() {
+    asm volatile("nop \n nop \n nop");
+    gpio_put(PIN_CS, 1);
+    asm volatile("nop \n nop \n nop");
+}
+
+static void wiznet_reset() {
+    // Two byte reset. First byte register, second byte data
+    // There are a load more options to set up the device in different ways that could be added here
+
+    gpio_set_dir(PIN_RST, GPIO_OUT);
+    gpio_put(PIN_RST, 0);
+    sleep_ms(100);
+    gpio_put(PIN_RST, 1);
+    sleep_ms(100);
+    bi_decl(bi_1pin_with_name(PIN_RST, "W5500 RST"));
+}
+
+uint8_t  wiznet_read(void) {
+    // For this particular device, we send the device the register we want to read
+    // first, then subsequently read from the device. The register is auto incrementing
+    // so we don't need to keep sending the register we want, just the first.
+    uint8_t rx = 0, tx = 0xFF;
+        spi_read_blocking(SPI_PORT, &tx,  &rx, 1);
```

Comment:
weird indentation

Reply:
I fixed indentation. Check the one more time, please